packages/valory/skills/test_solana_tx_abci/rounds.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2023 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the data classes for the Hello World ABCI application."""
+import json
+from enum import Enum
+from typing import Dict, Optional, Tuple, cast, List, Any
+
+from packages.valory.skills.abstract_round_abci.base import (
+    AbciApp,
+    AbciAppTransitionFunction,
+    AppState,
+    BaseSynchronizedData,
+    CollectSameUntilThresholdRound, get_name, DegenerateRound,
+)
+from packages.valory.skills.test_solana_tx_abci.payloads import SolanaTransactionPayload
+
+
+class Event(Enum):
+    """Event enumeration for the solana test skill."""
+
+    DONE = "done"
+    ROUND_TIMEOUT = "round_timeout"
+    ERROR = "error"
+    NO_MAJORITY = "no_majority"
+
+class SynchronizedData(BaseSynchronizedData):
+    """
+    Class to represent the synchronized data.
+
+    This data is replicated by the tendermint application.
+    """
+
+    @property
+    def most_voted_instruction_set(self) -> List[Dict[str, Any]]:
+        """Return the list of instructions selected for execution on the multisig."""
+        return cast(
+            List[Dict[str, Any]], self.db.get_strict("most_voted_instruction_set")
+        )
+
+
+class SolanaRound(CollectSameUntilThresholdRound):
+    """Dummy solana round."""
+
+    payload_class = SolanaTransactionPayload
+    synchronized_data_class = BaseSynchronizedData
+
+    @staticmethod
+    def _decode_instructions(instructions: str) -> List[Any]:
+        """Decode the instructions."""
+        instructions = json.loads(instructions)
+        return [json.loads(instruction) for instruction in instructions]
```

Comment:
@Adamantios suggests we should store this in a serialized form. It will make such rounds much simpler in the future. I agree with that suggestion.

Reply:
@angrybayblade ^

Reply:
So I tried to address this here https://github.com/valory-xyz/open-autonomy/pull/2139/commits/d2a88ad866495f285421c60cfe90c6bf56e972ed

The contract method was returning a serialised instruction string  instead of a dictionary object, I updated the transfer ix method to return a dictionary to match the rest of the methods